runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -2301,22 +2422,65 @@ J9::Z::TreeEvaluator::instanceofEvaluator(TR::Node * node, TR::CodeGenerator * c
       }
    else
       {
-      return TR::TreeEvaluator::VMinstanceOfEvaluator(node, cg);
+      static bool initialResult = feGetEnv("TR_instanceOfInitialValue") != NULL;
+      traceMsg(comp,"Initial result = %d\n",initialResult);
+      // Complementing Initial Result to True if the floag is not passed.
+      return VMgenCoreInstanceofEvaluator(node,cg,NULL,NULL,!initialResult,1,NULL,false);
       }
    }
 
-static void
-generateNullChkSnippet(
-      TR::Node *node,
-      TR::CodeGenerator *cg)
+/** \brief
+ *     Generates null test of \p objectReg for instanceof or checkcast[AndNULLCHK] \p node. In case a NULLCHK is
+ *     required this function will generate the sequence which throws the appropriate exception.
+ *
+ *  \param node
+ *     The instanceof, checkcast, or checkcastAndNULLCHK node.
+ *
+ *  \param cg
+ *     The code generator used to generate the instructions.
+ *
+ *  \param objectReg
+ *     The object to null test.
+ *
+ *  \return
+ *     \c true if a boolean condition code is set and the callee is expected to act on it; \c false otherwise, meaning
+ *     a NULLCHK was performed and if \p objectReg was null an exception throwing fallback path will be taken.
+ */
+static bool
+genInstanceOfOrCheckCastNullTest(TR::Node* node, TR::CodeGenerator* cg, TR::Register* objectReg)
    {
-   TR::Compilation *comp = cg->comp();
-   TR::LabelSymbol * snippetLabel = generateLabelSymbol(cg);
-   TR::S390BranchInstruction * brInstr = (TR::S390BranchInstruction*) generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BE, node, snippetLabel);
-   brInstr->setExceptBranchOp();
+   if (node->getOpCodeValue() == TR::checkcastAndNULLCHK)
+      {
+      if (cg->getHasResumableTrapHandler())
+         {
+         TR::Instruction* compareAndTrapInsturction = generateRIEInstruction(cg, TR::InstOpCode::getCmpImmTrapOpCode(), node, objectReg, 0, TR::InstOpCode::COND_BE);
+         compareAndTrapInsturction->setExceptBranchOp();
```

Comment:
Minor typo in the variable name here, but I think this is pre-existing code.  Can be fixed in a subsequent PR so that testing doesn't need to be restarted.